Makefile

```diff
@@ -1,198 +1,336 @@
 #!/usr/bin/make -f
 
-PREFIX=/opt/yandex/ch-tools
-INSTALL_DIR=$(DESTDIR)$(PREFIX)
+ifndef VERBOSE
+.SILENT:
+endif
+
+SHELL := bash
+.ONESHELL:
+.SHELLFLAGS := -eu -o pipefail -c
+.DELETE_ON_ERROR:
+MAKEFLAGS += --warn-undefined-variables
+MAKEFLAGS += --no-builtin-rules
+
+PYTHON ?= python3
+
+# The latest version supporting Python 3.6
+POETRY_VERSION ?= 1.1.15
+POETRY_HOME ?= /opt/poetry
+POETRY := $(POETRY_HOME)/bin/poetry
+
+PREFIX ?= /opt/yandex/ch-tools
+BUILD_PYTHON_OUTPUT_DIR ?= dist
+BUILD_DEB_OUTPUT_DIR ?= out
+SRC_DIR ?= ch_tools
+TESTS_DIR ?= tests
+
+# It is used by DEB building tools to install a program to temporary
+# directory before packaging
+DESTDIR ?=
+
+INSTALL_DIR = $(DESTDIR)$(PREFIX)
+BIN_DIR = $(INSTALL_DIR)/bin
+SYMLINK_BIN_DIR = $(DESTDIR)/usr/bin
+
+WHL_FILE = ch_tools-*.whl
+VENV_DIR = .venv
+VERSION_FILE = version.txt
+INSTALL_DEPS_STAMP = .install-deps
+
+# Pass arguments for testing tools
+BEHAVE_ARGS ?= -D skip_setup
+PYTEST_ARGS ?=
+
 
 .PHONY: install
 install: install-python-package install-symlinks install-bash-completions configure-logs ;
 
+
 .PHONY: uninstall
 uninstall: uninstall-python-package uninstall-symlinks uninstall-bash-completions uninstall-logrotate ;
 
+
+.PHONY: install-deps
+install-deps: $(INSTALL_DEPS_STAMP) ;
+
+
+# Update dependencies in poetry.lock to their latest versions according to your pyproject.toml
+.PHONY: update-deps
+update-deps:
+	$(POETRY) update
+
+
+$(INSTALL_DEPS_STAMP): ensure-poetry venv pyproject.toml 
+	$(POETRY) install --no-root
+	touch $(INSTALL_DEPS_STAMP)
+
+.PHONY: install-poetry
+install-poetry:
+	if [ ! -e $(POETRY) ]; then
+		echo "Installing poetry $(POETRY_VERSION)..."
+		curl -sSL https://install.python-poetry.org | POETRY_HOME=$(POETRY_HOME) $(PYTHON) - --version $(POETRY_VERSION)
+
+		# Fix cannot "import name 'appengine' from 'urllib3.contrib'..." error 
+		# while 'poetry publish' for version poetry 1.1.15
+		# https://urllib3.readthedocs.io/en/stable/v2-migration-guide.html#importerror-cannot-import-name-gaecontrib-from-requests-toolbelt-compat
+		$(POETRY_HOME)/venv/bin/python -m pip install urllib3==1.26.15
+	else
+		echo "Found installed poetry $$($(POETRY) --version)"				
+	fi
```

Comment:
Wouldn't it be easier to install poetry using `pip` and `venv`?

https://gist.githubusercontent.com/Alex-Burmak/ca4f748a9673748707205e4fd62752f1/raw/f6c05e6ffaab96d190b4fc5742abaad9a7b7ab93/venv.txt

It can be either the same venv as for other dependencies or dedicated venv for poetry.

Reply:
Yes it is quite possible. I did it via pip in a separate venv because mixing project's and poetry's dependencies in one environment is a bad idea.